airflow/providers/sqlite/operators/sqlite.py

```diff
@@ -49,6 +50,6 @@ def __init__(self, *, sqlite_conn_id: str = "sqlite_default", **kwargs) -> None:
         warnings.warn(
             """This class is deprecated.
             Please use `airflow.providers.common.sql.operators.sql.SQLExecuteQueryOperator`.""",
-            DeprecationWarning,
+            AirflowProviderDeprecationWarning,
```

Comment:
Not sure about this one.
sqlite provider doesn't have min Airflow version (as it's one of the pre installed providers)
Nothing blocks users from updating this provider independently with older version of Airflow which may result in import error.

I wonder if we have a way to prevent PIP from allowing it?

Reply:
Theorethically we could add dependency on airflow for pip (we used to have them) but it breaks tools that detect it as cycles, what we could do is that we could do similar thing as some other packages do - check it at runtime when it is used and raise a warning or error. This way if someone install a new version of sqlite on old airflow version it would refuse to start. 

It won't prevent it from being upgraded, but at least we can raise a clear error message in this case and ask the user to upgrade airflow if they want to update the provider to a newer version. 

I think we should do it anyway for all preinstalled packages - WDYT @eladkal ?

Reply:
yeah that is a good idea to do anyway!

My question here is to raise the two options:
1. Since our policy states that all providers are 2.4 compatible leave this as is and if users will face it then they will have to downgrade - Not user friendly and I hope to avoid this!
2. Add try/catch to support both imports thus being compatible with older versions of Airflow. This is kinda against our policy but this prevents the potential issue.

Reply:
I'd be for raising hard error. One of the reasons for having min-airflow version is to be able to AVOID this kind of hacks - even if theorethically the provider could work with older airflow. This is also a way to have our users more-or-less-gently pushed to upgrade airflow.

Reply:
that is reasonable.
Can this be done in the provider `__init.py__` ?
Something like:
```
    from airflow.version import version
    if Version(version) < Version("2.4"):
       raise ....
```

Reply:
Yes. That was very close to what I implemented in https://github.com/apache/airflow/pull/30994

Reply:
cool. Rebased after https://github.com/apache/airflow/pull/30994
lets see